src/libraries/System.Private.Xml/src/System/Xml/Core/LocalAppContextSwitches.cs

```diff
@@ -1,60 +1,45 @@
 
-using System.Runtime.CompilerServices;
-
-namespace System
+namespace System.Xml
 {
-    internal static partial class LocalAppContextSwitches
+    internal static class LocalAppContextSwitches
     {
-        private static int s_dontThrowOnInvalidSurrogatePairs;
-        public static bool DontThrowOnInvalidSurrogatePairs
-        {
-            [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            get
-            {
-                return GetCachedSwitchValue("Switch.System.Xml.DontThrowOnInvalidSurrogatePairs", ref s_dontThrowOnInvalidSurrogatePairs);
-            }
-        }
+        public static bool DontThrowOnInvalidSurrogatePairs { get; } =
+            AppContext.TryGetSwitch(
+                switchName: "Switch.System.Xml.DontThrowOnInvalidSurrogatePairs",
+                isEnabled: out bool value)
+            ? value : false;
+
+        public static bool IgnoreEmptyKeySequences { get; } =
+            AppContext.TryGetSwitch(
+                switchName: "Switch.System.Xml.IgnoreEmptyKeySequences",
+                isEnabled: out bool value)
+            ? value : false;
+
+        public static bool IgnoreKindInUtcTimeSerialization { get; } =
+            AppContext.TryGetSwitch(
+                switchName: "Switch.System.Xml.IgnoreKindInUtcTimeSerialization",
+                isEnabled: out bool value)
+            ? value : false;
 
-        private static int s_ignoreEmptyKeySequences;
-        public static bool IgnoreEmptyKeySequences
-        {
-            [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            get
-            {
-                return GetCachedSwitchValue("Switch.System.Xml.IgnoreEmptyKeySequencess", ref s_ignoreEmptyKeySequences);
-            }
-        }
+        public static bool LimitXPathComplexity { get; } =
+            AppContext.TryGetSwitch(
+                switchName: "Switch.System.Xml.LimitXPathComplexity",
+                isEnabled: out bool value)
+            ? value : false;
 
-        private static int s_ignoreKindInUtcTimeSerialization;
-        public static bool IgnoreKindInUtcTimeSerialization
-        {
-            [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            get
-            {
-                return GetCachedSwitchValue("Switch.System.Xml.IgnoreKindInUtcTimeSerialization", ref s_ignoreKindInUtcTimeSerialization);
-            }
-        }
+        public static bool AllowDefaultResolver { get; } =
+            AppContext.TryGetSwitch(
+                switchName: "Switch.System.Xml.AllowDefaultResolver",
+                isEnabled: out bool value)
+            ? value : false;
 
-        private static int s_limitXPathComplexity;
-        public static bool LimitXPathComplexity
-        {
-            [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            get
-            {
-                return GetCachedSwitchValue("Switch.System.Xml.LimitXPathComplexity", ref s_limitXPathComplexity);
```

Comment:
BTW: The previous patten was a bit more trimming friendly. For example, this call to fetch "Switch.System.Xml.LimitXPathComplexity" can be completely trimmed out in the existing code when the app does not use XPath. It will be always kept and executed eagerly in the new code.